cloudferrylib/os/network/neutron.py

```diff
@@ -546,20 +546,16 @@ def convert_security_groups(sec_gr, cloud):
         identity_res = cloud.resources[utl.IDENTITY_RESOURCE]
         net_res = cloud.resources[utl.NETWORK_RESOURCE]
 
-        get_tenant_name = identity_res.get_tenants_func()
-
-        security_group_rules = []
-        for rule in sec_gr['security_group_rules']:
-            rule_info = NeutronNetwork.convert(rule, cloud, 'rule')
-            security_group_rules.append(rule_info)
+        tenants = identity_res.get_tenants_list_dict()
 
         result = {
             'name': sec_gr['name'],
             'id': sec_gr['id'],
             'tenant_id': sec_gr['tenant_id'],
-            'tenant_name': get_tenant_name(sec_gr['tenant_id']),
+            'tenant_name': tenants[sec_gr['tenant_id']],
```

Comment:
Looks like you re-wrote get_tenants_func(), but you don't use it anymore.
     def get_tenants_func(self):
         tenants = self.get_tenants_list_dict()

Anyway... 
tenants[sec_gr['tenant_id']]
seems to fail if there is no such tenant ID...
I would suggest to use something like this:
tenants.get(sec_gr['tenant_id'], 'admin')
 
         def func(tenant_id):
             return tenants.get(tenant_id, 'admin')
 
         return func